lxd/resources/network_ethtool.go

```diff
@@ -219,6 +247,108 @@ func ethtoolGset(ethtoolFd int, req *ethtoolReq, info *api.ResourcesNetworkCardP
 	return nil
 }
 
+func ethtoolLink(ethtoolFd int, req *ethtoolReq, info *api.ResourcesNetworkCardPort) error {
+	// Interface info
+	ethLinkSettings := ethtoolLinkSettings{
+		cmd: 0x0000004c,
+	}
+	req.data = uintptr(unsafe.Pointer(&ethLinkSettings))
+
+	// Retrieve size of masks
+	_, _, errno := unix.Syscall(unix.SYS_IOCTL, uintptr(ethtoolFd), unix.SIOCETHTOOL, uintptr(unsafe.Pointer(req)))
+	if errno != 0 {
+		return errors.Wrap(unix.Errno(errno), "Failed to ETHTOOL_GLINKSETTINGS")
+	}
+
+	// This insane interface gives us the size of the masks as a negative value.
+	if ethLinkSettings.linkModeMasksNwords >= 0 || ethLinkSettings.cmd != 0x0000004c {
+		return errors.Wrap(unix.Errno(unix.EINVAL), "Failed to ETHTOOL_GLINKSETTINGS")
+	}
+
+	ethLinkSettings.cmd = 0x0000004c
```

Comment:
Why does it need to be set again?

Reply:
Not sure if the kernel clobbers it when it does a `copy_to_user()`. So better safe.

Reply:
Wait, it can't.